File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/java/CesarCarmona30.java
// Sitio web del lenguaje:
// https://docs.oracle.com/en/java/

// Comentario de línea

/**
 * Este es un comentario
 * de varias líneas
 */

/*
  Esto también es un
  comentario de
  varias líneas
 */

public class CesarCarmona30 {
  public static void main(String[] args) {
    
    // Variable
    String name = "Cesar";
    // Constante: se declaran con final
    final String GENDER = "MALE";

    // TIpos de datos primitivos
    byte numByte = 127;
    short numShort = 32767;
    int numInt = 2147483647;
    long numLong = 9223372036854775807L;
    float numFloat = 21.4f;
    double height = 1.75;
    boolean alive = true;
    boolean dead = false;
    char blood = 'O';
    String country = "México";

    String language = "Java";
    System.out.println("¡Hola, " + language + "!");
  }
}

File: Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/javascript/CesarCarmona30.js
// Sitio web del lenguaje
// https://developer.mozilla.org/es/docs/Web/JavaScript

// Formas de elaborar comentarios

// Comentario de línea

/*
  Comentario multilínea
 */

/**
 * Esto también es un
 * comentario multilínea
 */

// Variables

/*
  var y let sirven para declarar variables
  var tiene un ámbito global o local,
  el hoisting con var permite mover
  la declaración en la cabecera del
  código inicializandolo en undefined
  let tiene un ámbito de bloque,
  el hoisting con let tambien mueve
  la declaración a la cabecera del 
  código pero no se inicializa
  var se encuentra en desuso, por
  buenas prácticas es mejor ocupar let
*/
var country = 'México';
let name = 'César';
// las constantes se manejan con const
const sex = 'Masculino';

// Variables para datos primitivos

// int
let age = 19;
// float
let height = 1.75;
// boolean
let alive = true;
let dead = false;
// string
let city = "México City";
// undefined
let status = undefined;
// Symbol
let my_symbol = Symbol('My symbol');

const language = 'JavaScript'
console.log(`¡Hola, ${language}!`);